From 34420734220724a509719a1dfd206796d899eb6e Mon Sep 17 00:00:00 2001
From: Jiang Lu <lu.jiang@windriver.com>
Date: Wed, 28 Jan 2015 15:54:17 +0800
Subject: [PATCH] T2080:Add 1533MHz rcw for T2080 r1.0

Add 1533MHz rcw for T2080 revision 1.0 silicon.
All rcw file extracted from SDK 1.6.

Signed-off-by: Jiang Lu <lu.jiang@windriver.com>
---
 t2080qds/RR_PNNPPH_66_15/rcw_66_15_1533MHz.rcw |   56 ++++++++++++++++++++++
 t2080qds/RR_PNSNNR_6C_2D/rcw_6c_2d_1533MHz.rcw |   58 +++++++++++++++++++++++
 t2080rdb/RRFFXX_P_66_15/rcw_66_15_1533MHz.rcw  |   55 ++++++++++++++++++++++
 t2080rdb/RRFFXX_P_66_1F/rcw_66_1f_1533MHz.rcw  |   60 ++++++++++++++++++++++++
 t2080rdb/RRFFXX_P_66_27/rcw_66_27_1533MHz.rcw  |   55 ++++++++++++++++++++++
 5 files changed, 284 insertions(+), 0 deletions(-)
 create mode 100644 t2080qds/RR_PNNPPH_66_15/rcw_66_15_1533MHz.rcw
 create mode 100644 t2080qds/RR_PNSNNR_6C_2D/rcw_6c_2d_1533MHz.rcw
 create mode 100644 t2080rdb/RRFFXX_P_66_15/rcw_66_15_1533MHz.rcw
 create mode 100644 t2080rdb/RRFFXX_P_66_1F/rcw_66_1f_1533MHz.rcw
 create mode 100644 t2080rdb/RRFFXX_P_66_27/rcw_66_27_1533MHz.rcw

diff --git a/t2080qds/RR_PNNPPH_66_15/rcw_66_15_1533MHz.rcw b/t2080qds/RR_PNNPPH_66_15/rcw_66_15_1533MHz.rcw
new file mode 100644
index 0000000..649cc18
--- /dev/null
+++ b/t2080qds/RR_PNNPPH_66_15/rcw_66_15_1533MHz.rcw
@@ -0,0 +1,56 @@
+/*
+ * T2080QDS RCW for SerDes Protocol PRTCL_S1=0x66, PRTCL_S2=0x15
+ *
+ * 42G configuration -- 2 RGMII + 4 XFI
+ *
+ * Frequencies:
+ *
+ * SYSCLK: 66.66 MHz        (SW3[1:4] = 0000)
+ * DDRCLK: 133.33MHz	    (SW3[5:6] = 11)
+ * SD1_REF1 CLK: 156.25 MHz (SW4[1:2] = 10)
+ * SD1_REF2 CLK: 100 MHz    (SW4[3:4] = 11)
+ * SD2_REF1 CLK: 100 MHz    (SW4[5:6] = 11)
+ * SD2_REF2 CLK: 100 MHz    (SW4[7:8] = 11)
+ *
+ * Core -- 1533 MHz (Mul 23)
+ * Platform -- 600 MHz (Mul 9)
+ * DDR -- 1066.7 MHz (2133 MT/s) (Mul 16)
+ * FMAN -- 700 MHz (HWA_CGA_M1_CLK_SEL=6)
+ *
+ * RGMII1: MAC3
+ * RGMII2: MAC4
+ * XFI:    XFI1, XFI2, XFI3, XFI4
+ * SATA:   SATA1 and SATA2
+ * SRIO:   NULL
+ *
+ * Slot  Card
+ * 1     PCIe4 x4
+ * 2     NULL
+ * 3     NULL
+ * 4     PCIe1 x4
+ * 5     PCIe2 x2
+ *
+ * PBI source is NOR
+ */
+
+#include <t2080.rcwi>
+
+SYS_PLL_RAT=9
+MEM_PLL_RAT=16
+CGA_PLL1_RAT=23
+CGA_PLL2_RAT=21
+HWA_CGA_M1_CLK_SEL=6
+HWA_CGA_M2_CLK_SEL=4
+FM_MAC_RAT=1
+SRDS_PRTCL_S1=0x66
+SRDS_PRTCL_S2=0x15
+SRDS_PLL_REF_CLK_SEL_S1=0
+SRDS_PLL_REF_CLK_SEL_S2=0
+SRDS_DIV_PEX_S1=2
+SRDS_DIV_PEX_S2=2
+PBI_SRC=14
+BOOT_LOC=24
+IFC_MODE=39
+DRAM_LAT=1
+UART_BASE=6
+IRQ_BASE=511
diff --git a/t2080qds/RR_PNSNNR_6C_2D/rcw_6c_2d_1533MHz.rcw b/t2080qds/RR_PNSNNR_6C_2D/rcw_6c_2d_1533MHz.rcw
new file mode 100644
index 0000000..9b101e7
--- /dev/null
+++ b/t2080qds/RR_PNSNNR_6C_2D/rcw_6c_2d_1533MHz.rcw
@@ -0,0 +1,58 @@
+/*
+ * T2080QDS RCW for SerDes Protocol PRTCL_S1=0x6C, PRTCL_S2=0x2D
+ *
+ * 24G configuration -- 2 RGMII + 2 SGMII + 2 XFI
+ *
+ * Frequencies:
+ *
+ * SYSCLK: 66.66 MHz        (SW3[1:4] = 0000)
+ * DDRCLK: 133.33MHz        (SW3[5:6] = 11)
+ * SD1_REF1 CLK: 156.25 MHz (SW4[1:2] = 10)
+ * SD1_REF2 CLK: 100 MHz    (SW4[3:4] = 11)
+ * SD2_REF1 CLK: 100 MHz    (SW4[5:6] = 11)
+ * SD2_REF2 CLK: 100 MHz    (SW4[7:8] = 11)
+ *
+ * Core -- 1533 MHz (Mul 23)
+ * Platform -- 600 MHz (Mul 9)
+ * DDR -- 1066.7 MHz (2133 MT/s) (Mul 16)
+ * FMAN -- 700 MHz (HWA_CGA_M1_CLK_SEL=6)
+ *
+ * RGMII1: MAC3
+ * RGMII2: MAC4
+ * XFI:    XFI1 and XFI2
+ * SATA:   NULL
+ * SRIO:   SRIO1, SRIO2
+ *
+ * Slot  Card
+ * 1     PCIe4 x4
+ * 2     NULL
+ * 3     SGMII
+ * 4     SRIO2
+ * 5     SRIO1
+ *
+ * PBI source is NOR
+ */
+
+#include <t2080.rcwi>
+
+SYS_PLL_RAT=9
+MEM_PLL_RAT=16
+CGA_PLL1_RAT=23
+CGA_PLL2_RAT=21
+HWA_CGA_M1_CLK_SEL=6
+HWA_CGA_M2_CLK_SEL=4
+FM_MAC_RAT=1
+SRDS_PRTCL_S1=0x6c
+SRDS_PRTCL_S2=0x2d
+SRDS_PLL_REF_CLK_SEL_S1=0
+SRDS_PLL_REF_CLK_SEL_S2=0
+SRDS_DIV_PEX_S1=1
+SRDS_DIV_PEX_S2=1
+PBI_SRC=14
+BOOT_LOC=24
+IFC_MODE=39
+DRAM_LAT=1
+UART_BASE=6
+IRQ_BASE=511
+
+#include "../../a007837.rcw"
diff --git a/t2080rdb/RRFFXX_P_66_15/rcw_66_15_1533MHz.rcw b/t2080rdb/RRFFXX_P_66_15/rcw_66_15_1533MHz.rcw
new file mode 100644
index 0000000..e9714ea
--- /dev/null
+++ b/t2080rdb/RRFFXX_P_66_15/rcw_66_15_1533MHz.rcw
@@ -0,0 +1,55 @@
+/*
+ * T2080RDB RCW for SerDes Protocol PRTCL_S1=0x66, PRTCL_S2=0x15
+ *
+ * 42G configuration:
+ *     - 2 RGMII ports
+ *     - 2 10G SFP+ ports
+ *     - 2 10G Base-T ports
+ *
+ * PEX4 Gen3 x4(slot) + PEX1 Gen2 x4(finger) + PEX2 Gen2 x2(C293) + SATA x2
+ *
+ * Frequencies:
+ *
+ * SYSCLK: 66.66 MHz
+ * DDRCLK: 133.33 MHz
+ * SD1_REF1 CLK: 156.25 MHz
+ * SD1_REF2 CLK: 100 MHz
+ * SD2_REF1 CLK: 100 MHz
+ * SD2_REF2 CLK: 100 MHz
+ *
+ * Core -- 1533 MHz (Mul 23)
+ * Platform -- 600 MHz (Mul 9)
+ * DDR -- 800 MHz (1600 MT/s) (Mul 12)
+ * FMAN -- 700 MHz (HWA_CGA_M1_CLK_SEL=6)
+ *
+ * RGMII1: MAC3
+ * RGMII2: MAC4
+ * 10G SFP+ ports:   MAC9 and MAC10
+ * 10G Base-T ports: MAC1 and MAC2
+ * SATA: SATA1 and SATA2
+ * PCIe4 x4 in the PCI Slot
+ *
+ */
+
+#include <../t2080qds/t2080.rcwi>
+
+SYS_PLL_RAT=9
+MEM_PLL_RAT=12
+CGA_PLL1_RAT=23
+CGA_PLL2_RAT=21
+HWA_CGA_M1_CLK_SEL=6
+HWA_CGA_M2_CLK_SEL=4
+FM_MAC_RAT=1
+SRDS_PRTCL_S1=0x66
+SRDS_PRTCL_S2=0x15
+SRDS_PLL_REF_CLK_SEL_S1=0
+SRDS_PLL_REF_CLK_SEL_S2=0
+SRDS_DIV_PEX_S1=0
+SRDS_DIV_PEX_S2=0
+HOST_AGT_PEX=4
+PBI_SRC=14
+BOOT_LOC=24
+IFC_MODE=39
+DRAM_LAT=1
+UART_BASE=6
+IRQ_BASE=511
diff --git a/t2080rdb/RRFFXX_P_66_1F/rcw_66_1f_1533MHz.rcw b/t2080rdb/RRFFXX_P_66_1F/rcw_66_1f_1533MHz.rcw
new file mode 100644
index 0000000..36937bf
--- /dev/null
+++ b/t2080rdb/RRFFXX_P_66_1F/rcw_66_1f_1533MHz.rcw
@@ -0,0 +1,60 @@
+/*
+ * T2080RDB RCW for SerDes Protocol PRTCL_S1=0x66, PRTCL_S2=0x1F
+ *
+ * 42G configuration:
+ *     - 2 RGMII ports
+ *     - 2 10G SFP+ ports
+ *     - 2 10G Base-T ports
+ *
+ * PEX4 Gen3 x4(slot) + PEX1 Gen3 x4(finger) + PEX2 Gen2 x2(C293), no SATA
+ *
+ * Frequencies:
+ *
+ * SYSCLK: 66.66 MHz
+ * DDRCLK: 133.33 MHz
+ * SD1_REF1 CLK: 156.25 MHz
+ * SD1_REF2 CLK: 100 MHz
+ * SD2_REF1 CLK: 100 MHz
+ * SD2_REF2 CLK: 100 MHz
+ *
+ * Core -- 1533 MHz (Mul 23)
+ * Platform -- 600 MHz (Mul 9)
+ * DDR -- 800 MHz (1600 MT/s) (Mul 12)
+ * FMAN -- 700 MHz (HWA_CGA_M1_CLK_SEL=6)
+ *
+ * RGMII1: MAC3
+ * RGMII2: MAC4
+ * 10G SFP+ ports:   MAC9 and MAC10
+ * 10G Base-T ports: MAC1 and MAC2
+ * SATA: no
+ * PCIe4 x4 in the PCI Slot
+ *
+ */
+
+#include <../t2080qds/t2080.rcwi>
+
+SYS_PLL_RAT=9
+MEM_PLL_RAT=12
+CGA_PLL1_RAT=23
+CGA_PLL2_RAT=21
+HWA_CGA_M1_CLK_SEL=6
+HWA_CGA_M2_CLK_SEL=4
+FM_MAC_RAT=1
+SRDS_PRTCL_S1=0x66
+SRDS_PRTCL_S2=0x1f
+SRDS_PLL_REF_CLK_SEL_S1=0
+SRDS_PLL_REF_CLK_SEL_S2=0
+SRDS_DIV_PEX_S1=0
+SRDS_DIV_PEX_S2=0
+HOST_AGT_PEX=4
+PBI_SRC=14
+BOOT_LOC=24
+IFC_MODE=39
+DRAM_LAT=1
+UART_BASE=6
+IRQ_BASE=511
+
+.pbi
+//errata A-007186 for serdes2=0x1F, set SRDS2PLL2CR1=0x80004100
+write 0x0eb028, 0x80004100
+.end
diff --git a/t2080rdb/RRFFXX_P_66_27/rcw_66_27_1533MHz.rcw b/t2080rdb/RRFFXX_P_66_27/rcw_66_27_1533MHz.rcw
new file mode 100644
index 0000000..b45450c
--- /dev/null
+++ b/t2080rdb/RRFFXX_P_66_27/rcw_66_27_1533MHz.rcw
@@ -0,0 +1,55 @@
+/*
+ * T2080RDB RCW for SerDes Protocol PRTCL_S1=0x66, PRTCL_S2=0x27
+ *
+ * 42G configuration:
+ *     - 2 RGMII ports
+ *     - 2 10G SFP+ ports
+ *     - 2 10G Base-T ports
+ *
+ * PEX4 Gen3 x4(slot) + PEX1 Gen3 x4(finger) + SATA x2, no C293
+ *
+ * Frequencies:
+ *
+ * SYSCLK: 66.66 MHz
+ * DDRCLK: 133.33 MHz
+ * SD1_REF1 CLK: 156.25 MHz
+ * SD1_REF2 CLK: 100 MHz
+ * SD2_REF1 CLK: 100 MHz
+ * SD2_REF2 CLK: 100 MHz
+ *
+ * Core -- 1533 MHz (Mul 23)
+ * Platform -- 600 MHz (Mul 9)
+ * DDR -- 800 MHz (1600 MT/s) (Mul 12)
+ * FMAN -- 700 MHz (HWA_CGA_M1_CLK_SEL=6)
+ *
+ * RGMII1: MAC3
+ * RGMII2: MAC4
+ * 10G SFP+ ports:   MAC9 and MAC10
+ * 10G Base-T ports: MAC1 and MAC2
+ * SATA: SATA1 and SATA2
+ * PCIe4 x4 in the PCI Slot
+ *
+ */
+
+#include <../t2080qds/t2080.rcwi>
+
+SYS_PLL_RAT=9
+MEM_PLL_RAT=12
+CGA_PLL1_RAT=23
+CGA_PLL2_RAT=21
+HWA_CGA_M1_CLK_SEL=6
+HWA_CGA_M2_CLK_SEL=4
+FM_MAC_RAT=1
+SRDS_PRTCL_S1=0x66
+SRDS_PRTCL_S2=0x27
+SRDS_PLL_REF_CLK_SEL_S1=0
+SRDS_PLL_REF_CLK_SEL_S2=0
+SRDS_DIV_PEX_S1=0
+SRDS_DIV_PEX_S2=0
+HOST_AGT_PEX=4
+PBI_SRC=14
+BOOT_LOC=24
+IFC_MODE=39
+DRAM_LAT=1
+UART_BASE=6
+IRQ_BASE=511
-- 
1.7.1

